src/app/store/effects/users.effects.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from "@angular/core";
+import { Effect, Actions, ofType } from "@ngrx/effects";
+import { UserActions, LoginUser, LoginUserSuccess, LogOffUser } from "../actions/users.actions";
+import { map, switchMap, tap, first } from "rxjs/operators";
+import { Router } from "@angular/router";
+import { AuthService } from 'src/app/core/services/auth.service';
+import { NoopAction } from '../actions/common.actions';
+
+@Injectable()
+export class UserEffects {
+    @Effect()
+    loginUser$ = this.actions$.pipe(
+        ofType<LoginUser>(UserActions.LoginUser),
+        map(action => action.payload),
+        switchMap(({ login, password }) => {
+            return this.authService.login({ login, password }).pipe(
+                first(),
```

Comment:
there is no need to take first as far as http response is an observable of single element